pkg/cache/file_based.go

```diff
@@ -1,58 +1,109 @@
 package cache
 
 import (
+	"bufio"
 	"fmt"
 	"os"
 	"path/filepath"
+	"strings"
 
 	"github.com/adrg/xdg"
-	"github.com/k8sgpt-ai/k8sgpt/pkg/util"
 )
 
 var _ (ICache) = (*FileBasedCache)(nil)
 
-type FileBasedCache struct{}
+type FileBasedCache struct {
+	noCache bool
+}
+
+func (f *FileBasedCache) IsCacheDisabled() bool {
+	return f.noCache
+}
 
-func (*FileBasedCache) Exists(key string) bool {
-	path, err := xdg.CacheFile(filepath.Join("k8sgpt", key))
+func (*FileBasedCache) Load(key string) (string, error) {
+	path, err := xdg.CacheFile(filepath.Join("k8sgpt", "analysis"))
 
 	if err != nil {
-		fmt.Fprintln(os.Stderr, "warning: error while testing if cache key exists:", err)
-		return false
+		return "", err
 	}
 
-	exists, err := util.FileExists(path)
-
+	file, err := os.Open(path)
 	if err != nil {
-		fmt.Fprintln(os.Stderr, "warning: error while testing if cache key exists:", err)
-		return false
+		if os.IsNotExist(err) {
+			// Créer le fichier s'il n'existe pas
+			file, err = os.Create(path)
+			if err != nil {
+				return "", err
+			}
+		} else {
+			return "", err
+		}
+	}
+	defer file.Close()
+
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
```

Comment:
Nitpick: 
Not too much of a problem, but if you use k8sgpt a lot locally it will get gradually worse, as we linearly search through the file. 

Also imho this logic is pretty complicated in comparison to just use the file system as a database.